### PR TITLE
[Core] Add PDB and PriorityClass to capx components

### PR DESCRIPTION
**Pull request by esierra-stratio**
### Managed

- Configure capi-controller-manager and capa-controller-manager objects with two replicas and scale down to zero replicas capi-kubeadm-control-plane y capi-kubeadm-bootstrap.
- Add `PodDisruptionBudget` , minAvailable 1, to capi-controller-manager and capa-controller-manager.
- Add kubernetes `PriorityClass`  `system-node-critical` to capi-controller-manager and capa-controller-manager.

### Unmanaged

- Configure capx objects with two replicas (capi-controller-manager, capx-controller-manager, capi-kubeadm-control-plane y capi-kubeadm-bootstrap).
- Add `PodDisruptionBudget` , minAvailable 1, to capx objects (capi-controller-manager, capx-controller-manager, capi-kubeadm-control-plane y capi-kubeadm-bootstrap).
- Add kubernetes `PriorityClass`  `system-node-critical` to capx objects (capi-controller-manager, capx-controller-manager, capi-kubeadm-control-plane y capi-kubeadm-bootstrap).

### Example

**AWS EKS**

- Replicas
```bash
kw get deploy -o custom-columns=NAME:.metadata.name,REPLICAS:.spec.replicas -Alcluster.x-k8s.io/provider
NAME                                            REPLICAS
capa-controller-manager                         2
capi-kubeadm-bootstrap-controller-manager       0
capi-kubeadm-control-plane-controller-manager   0
capi-controller-manager                         2
```
- PodDisruptionBudget
```bash
kw get pdb -o custom-columns=NAME:.metadata.name,MINREPLICAS:.spec.minAvailable,CURRENTREPLICAS:.status.currentHealthy -Alcluster.x-k8s.io/provider
NAME                      MINREPLICAS   CURRENTREPLICAS
capa-controller-manager   1             2
capi-controller-manager   1             2
```
- PriorityClass
```bash
kw get deploy -o custom-columns=NAME:.metadata.name,PRIORITY:.spec.template.spec.priorityClassName -Alcluster.x-k8s.io/provider 
NAME                                            PRIORITY
capa-controller-manager                         system-node-critical
capi-kubeadm-bootstrap-controller-manager       <none>
capi-kubeadm-control-plane-controller-manager   <none>
capi-controller-manager                         system-node-critical

```
**AWS VMs**
- Replicas
```bash
kw get deploy -o custom-columns=NAME:.metadata.name,REPLICAS:.spec.replicas -Alcluster.x-k8s.io/provider
NAME                                            REPLICAS
capa-controller-manager                         2
capi-kubeadm-bootstrap-controller-manager       2
capi-kubeadm-control-plane-controller-manager   2
capi-controller-manager                         2
```
- PodDisruptionBudget
```bash
kw get pdb -o custom-columns=NAME:.metadata.name,MINREPLICAS:.spec.minAvailable,CURRENTREPLICAS:.status.currentHealthy -Alcluster.x-k8s.io/provider
NAME                                            MINREPLICAS   CURRENTREPLICAS
capa-controller-manager                         1             2
capi-kubeadm-bootstrap-controller-manager       1             2
capi-kubeadm-control-plane-controller-manager   1             2
capi-controller-manager                         1             2
```
- PriorityClass
```bash
kw get deploy -o custom-columns=NAME:.metadata.name,PRIORITY:.spec.template.spec.priorityClassName -Alcluster.x-k8s.io/provider
NAME                                            PRIORITY
capa-controller-manager                         system-node-critical
capi-kubeadm-bootstrap-controller-manager       system-node-critical
capi-kubeadm-control-plane-controller-manager   system-node-critical
capi-controller-manager                         system-node-critical
```

### CODE REVIEW

File: pkg/cluster/internal/create/actions/createworker/provider.go [hunk 1]
```diff
@@ -476,6 +498,93 @@ func (p *Provider) installCAPXWorker(n nodes.Node, kubeconfigPath string, allowA
 		return errors.Wrap(err, "failed to install CAPX in workload cluster")
 	}
 
+	// Manually assign PriorityClass to capa service
+	c = "kubectl --kubeconfig " + kubeconfigPath + " -n " + p.capxName + "-system patch deploy " + p.capxName + "-controller-manager -p '{\"spec\": {\"template\": {\"spec\": {\"priorityClassName\": \"system-node-critical\"}}}}' --type=merge"
+	_, err = commons.ExecuteCommand(n, c)
+	if err != nil {
+		return errors.Wrap(err, "failed to assigned priorityClass to capa-controller-manager")
```
Comment (stg-0) [inline pkg/cluster/internal/create/actions/createworker/provider.go:505]:
```suggestion
		return errors.Wrap(err, "failed to assigned priorityClass to " + p.capxName +"-controller-manager")
```

File: pkg/cluster/internal/create/actions/createworker/provider.go [hunk 2]
```diff
@@ -476,6 +498,93 @@ func (p *Provider) installCAPXWorker(n nodes.Node, kubeconfigPath string, allowA
...
+	}
+	c = "kubectl --kubeconfig " + kubeconfigPath + " -n " + p.capxName + "-system rollout status deploy " + p.capxName + "-controller-manager --timeout 60s"
+	if err != nil {
+		return errors.Wrap(err, "failed to check rollout status for capa-controller-manager")
```
Comment (stg-0) [inline pkg/cluster/internal/create/actions/createworker/provider.go:509]:
```suggestion
		return errors.Wrap(err, "failed to check rollout status for " + p.capxName +"-controller-manager")
```

File: pkg/cluster/internal/create/actions/createworker/provider.go [hunk 3]
```diff
@@ -476,6 +498,93 @@ func (p *Provider) installCAPXWorker(n nodes.Node, kubeconfigPath string, allowA
...
+
+	// Scale CAPX to 2 replicas
+	c = "kubectl --kubeconfig " + kubeconfigPath + " -n " + p.capxName + "-system scale --replicas 2 deploy " + p.capxName + "-controller-manager"
+	_, err = commons.ExecuteCommand(n, c)
+	if err != nil {
+		return errors.Wrap(err, "failed to scale CAPX in workload cluster")
+	}
+	c = "kubectl --kubeconfig " + kubeconfigPath + " -n " + p.capxName + "-system rollout status deploy " + p.capxName + "-controller-manager --timeout 60s"
+	if err != nil {
+		return errors.Wrap(err, "failed to check rollout status for capa-controller-manager")
```
Comment (stg-0) [inline pkg/cluster/internal/create/actions/createworker/provider.go:520]:
```suggestion
		return errors.Wrap(err, "failed to check rollout status for " + p.capxName +"-controller-manager")
```

**Review by stg-0 — CHANGES_REQUESTED** [review]

**Review by dandres-stratio — APPROVED** [review]
LGTM

Tested in EKS and GCP

**Review by stg-0 — APPROVED** [review]
LGTM

**Pull request merged**